src/lib/cargoContractInfo.ts

```diff
@@ -0,0 +1,13 @@
+export interface CargoContractInkDependency {
+  minCargoContractVersion: string;
+  validInkVersionRange: string;
+}
+
+// Keep cargo-contract versions in descending order
+// Ranges are supported by semver
+export const CARGO_CONTRACT_INK_DEPS: CargoContractInkDependency[] = [
+  { minCargoContractVersion: "4.0.0", validInkVersionRange: "<99.0.0" }, // Non-max version known yet: a very high version is used as fallback in the meantime
+  { minCargoContractVersion: "2.2.0", validInkVersionRange: "<5.0.0" },
```

Comment:
why do you skip `3.x` versions?

Reply:
According to the latest change logs, [cargo-contract `v2.2.0`](https://github.com/paritytech/cargo-contract/releases/tag/v2.2.0) specifies updating ink! to a minimum version of `v4.2.0`. While cargo-contract versions `v3.x` remain supported, they do not introduce new minimum requirements for ink!, thus maintaining the requirement for ink! `v4.2.0`.